src/net_processing.cpp

```diff
@@ -2795,6 +2845,7 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
                 // Do not process banned/discouraged addresses beyond remembering we received them
                 continue;
             }
+            num_proc++;
```

Comment:
This was `if (!peer->m_getaddr_sent) num_proc++;` in a previous incarnation and I thought it was ok. With unconditional `num_proc++`, observing addr/sec (`(addr_processed + addr_rate_limited) / connection_duration`) in order to find a better value for `MAX_ADDR_RATE_PER_SECOND` is not appropriate because the calculated addr/sec may be allowed to go higher than 0.1 without causing any rate limiting (we increment the tokens by 1000 when we send `getaddr`). I guess either way is fine.

About putting this increment after two possible `continue`s: I think that if we do some processing (`MayHaveUsefulAddressDB()` or banmap lookup) and decide to `continue`, that should be counted as "processing" too. I.e. `num_proc++;` should be put before `MayHaveUsefulAddressDB()` above. Or if it remains here, then maybe `addr_accepted` or `addr_added` is a name that would better describe what we have done. Maybe have both `addr_received` (`vAddr.size()`) and `addr_accepted`?

Reply:
@vasild I removed the "!peer->m_getaddr_sent" because it doesn't work reliably. If we receive a randomly relayed addr before receiving the response to GETADDR, the getaddr response will still be counted (and I see that happen fairly regularly).